src/posix/platform/platform-posix.h

```diff
@@ -100,8 +105,85 @@ enum
      * Unexpected radio spinel reset.
      */
     OT_EXIT_RADIO_SPINEL_RESET = 4,
+
+    /**
+     * System call or library function error.
+     */
+    OT_EXIT_ERROR_ERRNO = 5,
 };
 
+/**
+ * This function converts an exit code into a string.
+ *
+ * @param[in]  aExitCode  An exit code.
+ *
+ * @returns  A string representation of an exit code.
+ *
+ */
+const char *otExitCodeToString(uint8_t aExitCode);
+
+/**
+ * This macro checks for the specified condition, which is expected to commonly be true,
+ * and both records exit status and terminates the program if the condition is false.
+ *
+ * @param[in]   aCondition  The condition to verify
+ * @param[in]   aExitCode   The exit code.
+ *
+ */
+#define VerifyOrDie(aCondition, aExitCode)                                                               \
+    do                                                                                                   \
+    {                                                                                                    \
+        if (!(aCondition))                                                                               \
+        {                                                                                                \
+            fprintf(stderr, "exit(%d): %s: %s\r\n", aExitCode, __func__, otExitCodeToString(aExitCode)); \
```

Comment:
Suggestion: The printed message for all the macors looks almost the same `"exit(<>): <func> ..."`. How about having different ones to indicates if this is a `VerifyOrDie` or `SuccessOrDie` or `DieNow`... ?

Reply:
I would prefer to add `__LINE__` to the message to distinguish different messages. Because if a function calls function `VerifyOrDie` for many times, we still don't know which one generates the message. 
Comments?

Reply:
:+1: ...agree, `__LINE__` would be even better